zonemaster/zonemaster-backend
# Load config only once and always log STDERR

## Purpose

This PR does two things.
1. It makes zm-rpcapi not reload its config file for each worker on FreeBSD.
2. It makes zm-rpcapi write STDERR output to the log file on Linux.

## Context

Fixes #813.

## Changes

The options to starman in zm-rpcapi.lsb and zm_rpcapi.bsd are updated to be the same. The order of the options is also updated to make it easier to compare the two files.

## How to test this PR

This used to work only on Linux but not on FreeBSD:
1. Restart zm-rpcapi to make sure you have a fresh daemon started.
2. In one terminal run `tail -fn0 zm-rpcapi.log`.
3. In another terminal run `zmb version_info`.
4. Make sure the messages "NOTICE - Loading config:" is only logged once.

This used to work only on FreeBSD but not on Linux:
1. Update lib/Zonemaster/Backend/RPCAPI.pm to include the line `say STDERR "HERE";` in the `version_info` procedure.
2. Restart zm-rpcapi to make sure the modified code is loaded.
3. In one terminal run `tail -fn0 zm-rpcapi.log`.
4. In another terminal run `zmb version_info`.
5. Make sure the messages "HERE" is shows up in the output from the log file.
